src/test/java/com/google/starfish/services/FavoriteNoteServiceTest.java

```diff
@@ -0,0 +1,300 @@
+  private final String newCourse = "cs1300";
+  private final String firstNewUserId = Integer.toString(Integer.parseInt(constants.REFERENCE_USER_ID) + 1);
+  private final String secondNewUserId = Integer.toString(Integer.parseInt(constants.REFERENCE_USER_ID) + 2);
+
+  private static final boolean runTests = Constants.TEST_DB_NAME.equals("starfish_test");
+
+  /** Clear DB and insert reference data */
+  @Before
+  public void prepare() throws Exception {
+    if(!runTests) throw new Exception("Wrong Test Database Name");
+    Operation operation = getBeforeTestOperation();
+    DbSetup dbSetup = new DbSetup(new DataSourceDestination(pool), operation);
+    dbSetup.launch();
+  }
+
+  /** Test to see if we can get all notes favorited by a single user */
+  @Test
+  public void testGettingFavoriteNotesByUserId() throws SQLException {
+    Note[] favoriteNotes = favoriteNoteService.getFavoriteNotesByUserId(pool, firstNewUserId);
+    assertEquals(favoriteNotes.length, 7);
+  }
+
+  /** Test to see if we can get the number of times a particular note has been favorited */
+  @Test
+  public void testGettingNumFavoritesByNoteId() throws SQLException {
+    long numFavoritesOfNoteOne = favoriteNoteService.getNumFavoritesByNoteId(pool, 1);
+    long numFavoritesOfNoteTwo = favoriteNoteService.getNumFavoritesByNoteId(pool, 2);
+    assertTrue(numFavoritesOfNoteOne == 1);
+    assertTrue(numFavoritesOfNoteTwo == 2);
+  }
+
+  /** Test to see if we can correctly retrieve the most trending notes today */
+  @Test
+  public void testGettingTrendingNotesToday() throws SQLException, Exception {
+    Object[][] trendingNotesToday = favoriteNoteService.getTrendingNotesBySchoolOrCourse(pool, Recency.TODAY, null, null);
+    // We should be getting all 7 notes irrespective of whether they are trending or not
+    if (trendingNotesToday.length != 7) throw new Exception("Retrieved wrong number of notes: " + trendingNotesToday.length);
+    int numTrendingNotes = 2;
+    Note mostTrendingNote = (Note) trendingNotesToday[0][0];
+    long numFavoritesMostTrendingNote = (long) trendingNotesToday[0][1];
+    Note leastTrendingNote = (Note) trendingNotesToday[numTrendingNotes - 1][0];
+    long numFavoritesLeastTrendingNote = (long) trendingNotesToday[numTrendingNotes - 1][1];
+    long numFavoritesFirstNoteWithNoFavorite = (long) trendingNotesToday[numTrendingNotes][1];
+    assertTrue(mostTrendingNote.getId() == 2);
+    assertTrue(numFavoritesMostTrendingNote == 2);
+    assertTrue(leastTrendingNote.getId() == 1);
+    assertTrue(numFavoritesLeastTrendingNote == 1);
+    assertTrue(numFavoritesFirstNoteWithNoFavorite == 0);
+  }
+
+  /** Test to see if we can correctly retrieve the most trending notes this week */
+  @Test 
+  public void testGettingTrendingNotesThisWeek() throws SQLException, Exception {
+    Object[][] trendingNotesThisWeek = favoriteNoteService.getTrendingNotesBySchoolOrCourse(pool, Recency.THIS_WEEK, null, null);
```

Comment:
I see a lot of similar logic at the start of each test. Could this logic be consolidated into a helper function?

Reply:
Definitely, I think a lot of it can be extracted into a helper function that sets instance variables of the entire test class and that's called in the Before decorated prepare statement.